tests/v2/validation/prime/prime_test.go

```diff
@@ -0,0 +1,81 @@
+package prime
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+const rancherBrand = "suse"
+
+type PrimeVersionTestSuite struct {
+	suite.Suite
+	session        *session.Session
+	client         *rancher.Client
+	isPrime        bool
+	rancherVersion string
+}
+
+func (t *PrimeVersionTestSuite) TearDownSuite() {
+	t.session.Cleanup()
+}
+
+func (t *PrimeVersionTestSuite) SetupSuite() {
+	testSession := session.NewSession(t.T())
+	t.session = testSession
+	primeConfig := new(Config)
+	config.LoadConfig(ConfigurationFileKey, primeConfig)
+	t.isPrime = primeConfig.IsPrime
+	t.rancherVersion = primeConfig.RancherVersion
+
+	client, err := rancher.NewClient("", t.session)
+	assert.NoError(t.T(), err)
+	t.client = client
+
+}
+
+func (t *PrimeVersionTestSuite) TestPrimeVersion() {
+	var http_url = "https://" + t.client.RancherConfig.Host + "/rancherversion"
```

Comment:
Caleb I'm suggestion is, I think this whole block where you're checking the rancher version, and rancher prime bool should be in an extension called GetRancherVersion or something like that. Not in the test itself